### ISSUE TITLE
Please implement the batching rule for torch.matrix_exp.

### ISSUE THREAD

**Issue by LongZ-A** [pytorch/pytorch#115992]
### 🚀 The feature, motivation and pitch

Matrix exponentials are extremely expensive to compute, yet very important in many computing and ML problems . It would be great if such computational expensive functions can support batch implementations. Many thanks in advance for those who can help with this issue!

### Additional context

This below is the warning message from pytorch kernel: <ipython-input-13-6cfa90bb24a5>:3: UserWarning: There is a performance drop because we have not yet implemented the batching rule for aten::matrix_exp. Please file us an issue on GitHub so that we can prioritize its implementation. (Triggered internally at C:\cb\pytorch_1000000000000\work\aten\src\ATen\functorch\BatchedFallback.cpp:84.)

**Comment by Xiao215** [comment]
Hi, can I give this issue a try? :)

**Comment by LongZ-A** [comment]
Many thanks to everyone for the quick response! @zou3519 @Xiao215.  Hopefully this will be successfully implemented soon 👍

**Comment by Xiao215** [comment]
BTW I believe this is a repro of the issue.

```python
import torch
t = torch.randn(900, 3, 3)
o = torch.vmap(torch.matrix_exp)(t)
```

**Comment by amk16** [comment]
Hey could I attempt this if its still open?

**Comment by zou3519** [comment]
Issue still open, please feel free to attempt.

**Comment by KarthikSundar2002** [comment]
I don't think this issue was resolved by the last PR, so I think I can have a look at it right?

**Comment by zou3519** [comment]
Feel free to attempt

**Comment by pawan1162k** [comment]
@zou3519 Is this open to work? I am begginer and looking forward to contribute!
